lib/matplotlib/projections/polar.py

```diff
@@ -680,19 +680,20 @@ def __init__(self, *args, **kwargs):
         self.sticky_edges.y.append(0)
 
     def _wrap_locator_formatter(self):
-        self.set_major_locator(RadialLocator(self.get_major_locator(),
-                                             self.axes))
-        self.isDefault_majloc = True
+        if not isinstance(self.get_major_locator(), RadialLocator):
+            is_default = self.isDefault_majloc
+            super().set_major_locator(
+                RadialLocator(self.get_major_locator(), self.axes))
+            self.isDefault_majloc = is_default
+
+    def set_major_locator(self, locator):
```

Comment:
I guess you could inline the logic of _wrap_locator_formatter here (`if not isinstance(locator, RadialLocator): locator = RadialLocator(locator, self.axes)`) which would save you from having to mess with isdefault_majloc?

Reply:
Yes. Done.